### ISSUE TITLE
How can I generate a CSR with alternative names?

### ISSUE THREAD

**Issue by GINCHER** [kjur/jsrsasign#220]
Hey. 
Is there a way to generate a CSR with alternative names?

**Comment by kjur** [comment]
In current version, CSR extension such as subjectAltName is not supported in [CSRUtil.newCSRPEM](https://kjur.github.io/jsrsasign/api/symbols/KJUR.asn1.csr.CSRUtil.html#.newCSRPEM). You may add extension support to it. I'll add its support near in future.

**Comment by kjur** [comment]
Otherwise you can make it by CertificationRequestInfo class as following. I found that SubjectAltName extension is needed to be implement:

```JavaScript
csri = new KJUR.asn1.csr.CertificateRequestInfo();
csir.setSubjectByParam({str: "/C=US/O=Test"});
csri.setSubjectPublicKeyByGetKey(pubKeyObj);
san = ...need to implement SubjectAltName extension...;
csri.extensionArray.append(san);
csr = new KJUR.asn1.csr.CertificationRequest({csrinfo: csri});
csr.sign("SHA256withRSA", prvKey);
csrPEM = csr.getPEMString();
```

**Comment by GINCHER** [comment]
OK. Done, kinda. 
Here is the `extSeq` in `CertificationRequestInfo` that will add alternative names:
```javascript
var extSeq = KJUR.asn1.ASN1Util.newObject({seq: [{seq: [{oid: "1.2.840.113549.1.9.14"}, {set: [{seq: [{seq: [{oid: "2.5.29.17"}, {octstr: {obj: {seq: [
	{tag: ["a2", true, {utf8str: {str: "google.com"}}]},
	{tag: ["a2", true, {utf8str: {str: "facebook.com"}}]},
	{tag: ["a2", true, {utf8str: {str: "wikipedia.com"}}]}
]}}}]}]}]}]}]});
```
Tought I couldn't find a way to add it to CertificationRequestInfo without editing the whole function. 

And I have another question. For some reason I'm getting sometimes a 2047 bit key while using `KEYUTIL.generateKeypair("RSA", 2048)`, is there a fix?

**Comment by kjur** [comment]
I've just released 6.2.3 today. It supports CSR extensionRequest. Please take a look:
https://kjur.github.io/jsrsasign/api/symbols/KJUR.asn1.csr.CSRUtil.html#.newCSRPEM

```JavaScript
pem = KJUR.asn1.csr.CSRUtil.newCSRPEM({
  subject: {str: '/C=US/O=Test/CN=example.com'},
  ext: [
    {subjectAltName: {array: [{dns: 'example.net'}]}
  ],
  sbjpubkey: pubKeyPEM,
  sigalg: "SHA256withRSA",
  sbjprvkey: prvKeyPEM
});
```
